src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -433,41 +433,102 @@ class VariantSampleMatrix(val hc: HailContext, val metadata: VSMMetadata,
 
   def take(n: Int): Array[UnsafeRow] = unsafeRowRDD().take(n)
 
-  def groupVariantsBy(keyExpr: String, aggExpr: String, singleKey: Boolean = false): VariantSampleMatrix = {
+  def groupSamplesBy(keyExpr: String, aggExpr: String): VariantSampleMatrix = {
+    val localRowType = rowType
+    val sEC = EvalContext(Map(Annotation.GLOBAL_HEAD -> (0, globalSignature),
+      "s" -> (1, sSignature),
+      Annotation.SAMPLE_HEAD -> (2, saSignature)))
+    val (keyType, keyF) = Parser.parseExpr(keyExpr, sEC)
+    sEC.set(0, globalAnnotation)
+
+    val keysBySample = sampleIds.zip(sampleAnnotations).map { case (s, sa) =>
+      sEC.set(1, s)
+      sEC.set(2, sa)
+      keyF()
+    }
+    val newKeys = keysBySample.toSet.toIndexedSeq
```

Comment:
toIndexedSeq is highly triggering to me even though it's immediately converted to a map. Use `toArray` instead!